<launch>

    <arg name="simulation" default="true"/>

    <!-- SIMULATION -->
    <group if="$(arg simulation)">

        <!-- RTAB -->
        <node name="SLAM" pkg="autovalet_system" type="slam.sh" args="true"/>

        <!-- MOVEBASE -->
        <node name="NAVIGATION" pkg="autovalet_system" type="nav.sh" args="true" />

        <!-- ARUCO -->
        <include file="$(find autovalet_gazebo)/launch/aruco_sim.launch">
            <arg name="markerId" value="7" />
            <arg name="markerSize" value="0.5" />
            <arg name="ref_frame" value="/base_link" />
        </include>

        <!-- CAMERA PLUGIN DEPTH REGISTRATION -->
        <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

        <node pkg="nodelet" type="nodelet" name="nodelet1" args="load depth_image_proc/register nodelet_manager">
            <remap from="depth/camera_info" to="/frontCamera/depth/camera_info"/>
            <remap from="rgb/camera_info" to="/frontCamera/color/camera_info"/>
            <remap from="depth/image_rect" to="/frontCamera/depth/image_raw"/>
        </node>

        <!-- STATE MACHINE -->
        <node name="STATE_MACHINE" pkg="autovalet_system" type="run_sm.sh" args="true" />
    </group>

<!--_________________________________________________________________________________________________-->

    <!-- HUSKY -->
    <group unless="$(arg simulation)">

        <!-- RTAB -->
        <node name="SLAM" pkg="autovalet_system" type="slam.sh" args="false"/>

        <!-- MOVEBASE -->
        <node name="NAVIGATION" pkg="autovalet_system" type="nav.sh" args="false" />

        <!-- ARUCO -->
        <include file="$(find autovalet_husky)/launch/aruco_hardware.launch"/>

        <!-- STATE MACHINE -->
        <node name="STATE_MACHINE" pkg="autovalet_system" type="run_sm.sh" args="false" />
    </group>
</launch>
